get recipe maps ingredients to the response

jhcarrington opened the pull request.

Commit:
get recipe maps ingredients to the response

Commit:
lint

snoberger commented:
your changes to the validation functions broke the create recipe endpoint. create recipe does not pass ingredients as id and amount, it passes them as name, amount, metric

Commit:
Recipe/pantry get/create/update bug fixes

Return name and metric for ingredients on recipe get
Fix update recipe to allow for name changes
Fix update pantry, ingredient must be plural

Commit:
Change amount to be a string and fix data checkers
